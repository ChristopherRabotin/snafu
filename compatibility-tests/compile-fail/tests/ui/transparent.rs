extern crate snafu;

use snafu::prelude::*;

type BoxError = Box<dyn std::error::Error>;

#[derive(Debug, Snafu)]
enum Error {
    #[snafu(transparent)]
    MissingSource {},

    #[snafu(transparent)]
    HasUserFields { source: BoxError, a: i32, b: i32 },

    #[snafu(transparent(false))]
    TransparentFalseDoesNothing { source: BoxError },

    #[snafu(transparent)]
    #[snafu(display("Oh snap!"))]
    NonsensicalDisplay { source: BoxError },
}

fn main() {}
