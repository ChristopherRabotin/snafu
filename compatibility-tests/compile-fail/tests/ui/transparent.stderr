error: `transparent` errors must not have context fields
  --> tests/ui/transparent.rs:13:39
   |
13 |     HasUserFields { source: BoxError, a: i32, b: i32 },
   |                                       ^^^^^^

error: `transparent` errors must not have context fields
  --> tests/ui/transparent.rs:13:47
   |
13 |     HasUserFields { source: BoxError, a: i32, b: i32 },
   |                                               ^^^^^^

error: `transparent(false)` attribute has no effect
  --> tests/ui/transparent.rs:15:13
   |
15 |     #[snafu(transparent(false))]
   |             ^^^^^^^^^^^^^^^^^^

error: `transparent` errors cannot have a display format because they delegate `Display` to their source
  --> tests/ui/transparent.rs:19:13
   |
19 |     #[snafu(display("Oh snap!"))]
   |             ^^^^^^^^^^^^^^^^^^^

error: `transparent` errors cannot have a display format because they delegate `Display` to their source
  --> tests/ui/transparent.rs:18:13
   |
18 |     #[snafu(transparent)]
   |             ^^^^^^^^^^^

error: `transparent` errors must have a source field
  --> tests/ui/transparent.rs:10:5
   |
10 |     MissingSource {},
   |     ^^^^^^^^^^^^^
